### ISSUE TITLE
tag </programlisting> was inserted before </highlight> parsing python / cpp file to xml

### ISSUE THREAD

**Issue by sdrgroup** [doxygen/doxygen#8518]
**Describe the bug**
Describe what you see that (you think) is wrong.
When I parse a python file,reflection_test.py, the output file reflection__test_8py.xml is error. The tag </programlisting> was inserted before </highlight></codeline> .
**Expected behavior**
Describe what you would have expected or think is correct.
The tag </programlisting> should be after </highlight></codeline>.
**Screenshots**
If useful, add screenshots to help explain your problem.
![2021-04-25 19-16-58](https://user-images.githubusercontent.com/12608702/115991459-13a6b980-a5fb-11eb-9cf2-900ccf9f2c00.png)

**To Reproduce**
Attach a self contained example that allows us to reproduce the problem.
Such an example typically exist of some source code (can be dummy code) and a doxygen configuration file used (you can strip it using `doxygen -s -u`). After you verified the example demonstrates the problem, put it in a zip (or tarball) and attach it to the bug report. Try to avoid linking to external sources, since they might disappear in the future.
[reflection_test.zip](https://github.com/doxygen/doxygen/files/6371793/reflection_test.zip)

**Version**
Mention the version of doxygen used (output of `doxygen --version`) and the platform on which you run doxygen (e.g. Windows 10, 64 bit). If you run doxygen under Linux please also mention the name and version of the distribution used (output of `lsb_release -a`) and mention if you compiled doxygen yourself or that you use a binary that comes with the distribution or from the doxygen website.  
1.9.1
**Stack trace**
If you encounter a crash and can build doxygen from sources yourself with debug info (`-DCMAKE_BUILD_TYPE=Debug`), a stack trace can be very helpful (especially if it is not possible to capture the problem in a small example that can be shared).

**Additional context**

**Comment by sdrgroup** [comment]
my os is ubuntu 20.04

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8519

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by sdrgroup** [comment]
there is a same error when parsing another .h file.

[wire_format_lite_inl.zip](https://github.com/doxygen/doxygen/files/6396376/wire_format_lite_inl.zip)
the error is as below:
![image](https://user-images.githubusercontent.com/12608702/116503974-eb180b80-a8e9-11eb-9836-aa905190207f.png)

**Comment by albert-github** [comment]
Normally it would be better to open a new issue as we see here that the input file giving a problem is a cpp related file and not python related. The result on the output is the same though.

The offending parts are (as far as I quickly saw):
around line 778
```
  target = value.MessageType_WorkAroundCppLookupDefect
      ::SerializeWithCachedSizesToArray(target);
```
and around line 788
```
  return value.MessageType_WorkAroundCppLookupDefect
      ::SerializeWithCachedSizesToArray(target);
```
when, as a workaround but probably not feasible due to coding guidelines, removing the `\n` it looks OK again.

**Comment by albert-github** [comment]
For the second problem (cpp related), I've just pushed a proposed patch, pull request #8524

**Comment by albert-github** [comment]
Code for second problem has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
